src/glutil.h

```diff
@@ -105,6 +107,46 @@ inline void glLoadMatrix(const Imath::M44f& m)
     glLoadMatrixf((GLfloat*)m[0]);
 }
 
+//------------------------------------------------------------------------------
+// Texture utility
+
+struct Texture
+{
+    Texture(const QImage& i)
+    :   image(i),
+        target(GL_TEXTURE_2D),
+        texture(0)
+    {
```

Comment:
Shouldn't this check `i.format() == QImage::Format_ARGB32` or some such, just as a precaution?  If it's not in the right format, `image = image.convertToFormat(...)` should fix that.

I had a read through the qt source code, and `QGLContext::bindTexture()` has a whole texture cache and everything, based off cache keys held by `QImage`.  It's a bit complex and magical though, so I think I prefer what you have here.